packages/flutter/lib/src/material/switch.dart

```diff
@@ -183,10 +213,46 @@ class _SwitchState extends State<Switch> with TickerProviderStateMixin {
       inactiveTrackColor: inactiveTrackColor,
       configuration: createLocalImageConfiguration(context),
       onChanged: widget.onChanged,
-      additionalConstraints: additionalConstraints,
+      additionalConstraints: BoxConstraints.tight(getSwitchSize(theme)),
       vsync: this,
     );
   }
+
+  Widget buildCupertinoSwitch(BuildContext context) {
+    final Size size = getSwitchSize(Theme.of(context));
+    return Container(
+      width: size.width,
+      height: size.height,
+      alignment: Alignment.center,
+      child: CupertinoSwitch(
+        value: widget.value,
+        onChanged: widget.onChanged,
+        activeColor: widget.activeColor,
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    switch (widget._switchType) {
```

Comment:
Add the inherited override mechanism we were talking about?

Reply:
Arguably every feature is opt-in initially. 

The feature developer for a screen may not be the same person maintaining the overall app's visual aesthetics. There isn't an easy way to defer this choice up the tree this way unless the developer responsible for global visuals does a global search and replace one by one.

Reply:
I don't think a battery of flags that optionally disable the platform adaptations is useful enough to balance the cost of the additional API.  If demand for such a thing surfaces, we can always add it.